src/BenchmarkDotNet/Toolchains/DotNetSdkProvider.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace BenchmarkDotNet.Toolchains
+{
+    public class DotNetSdkProvider : IDotNetSdkProvider
+    {
+        private readonly string _customDotNetCliPath;
+
+        public string CustomDotNetCliPath => _customDotNetCliPath;
+
+        public DotNetSdkProvider(string customDotNetCliPath = null)
+        {
+            _customDotNetCliPath = customDotNetCliPath;
+        }
+
+        public IEnumerable<string> GetInstalledSdks()
+        {
+            var installedDotNetSdks = GetInstalledDotNetSdk();
+            var installedFrameworkSdks = GetInstalledFrameworkSdks();
+            return installedDotNetSdks.Concat(installedFrameworkSdks);
+        }
+
+        private IEnumerable<string> GetInstalledDotNetSdk()
+        {
+            string dotnetExecutable = string.IsNullOrEmpty(CustomDotNetCliPath) ? "dotnet" : CustomDotNetCliPath;
+            var startInfo = new ProcessStartInfo(dotnetExecutable, "--list-sdks")
+            {
+                RedirectStandardOutput = true,
+                UseShellExecute = false,
+                CreateNoWindow = true
+            };
+
+            using (var process = Process.Start(startInfo))
+            {
+                if (process == null) throw new InvalidOperationException("Failed to start the dotnet process.");
+                process.WaitForExit();
+
+                var output = process.StandardOutput.ReadToEnd();
+                var lines = output.Split(new[] { '\r', '\n' }, StringSplitOptions.RemoveEmptyEntries);
+                return lines.Select(line => line.Split(' ')[0]); // The SDK version is the first part of each line.
+            }
+        }
+
+        private IEnumerable<string> GetInstalledFrameworkSdks()
```

Comment:
Perhaps this should be pulled out to a separate `DotNetFrameworkSdkProvider`. It should only be necessary for `CsProjClassicNetToolchain`.